sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala

```diff
@@ -382,4 +385,6 @@ class AccountStateView(private val metadataStorageView: AccountStateMetadataStor
 
   override def getStateDbHandle: ResourceHandle = stateDb
 
+  override def getTransactionBlockNumber(txId: scorex.util.ModifierId): Int = metadataStorageView.getTransactionBlockNumber(txId).get
```

Comment:
I think, the result type must be as in Metadatastorage: `Option[Int]`,
otherwise we will throw an exception in case tx is not known

Reply:
My reasoning was that using an Option would have been like saying it is possible that a transaction doesn't have a block number, while in my understanding that shouldn't be possible. But in the end it is probably better using an Option and avoiding the exception.
Done